icu4j/maven-build/maven-icu4j-charset/pom.xml

```diff
@@ -0,0 +1,240 @@
+            <id>add-source</id>
+            <goals>
+              <goal>add-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>../../main/classes/charset/src</source>
+              </sources>
+            </configuration>
+          </execution>
+          
+          <!-- add directories with test code -->
+          <execution>
+            <id>add-test-source</id>
+            <goals>
+              <goal>add-test-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>../../main/tests/charset/src</source>
+                <source>../../main/tests/framework/src</source>
+              </sources>
+            </configuration>
+          </execution>
+
+          <!-- add directories with main executable resources -->
+          <execution>
+            <id>add-resource</id>
+            <goals>
+              <goal>add-resource</goal>
+            </goals>
+            <configuration>
+              <resources>
+                <resource>
+                  <directory>../../main/classes/charset/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+
+          <!-- add directories with test resources -->
+          <execution>
+            <id>add-test-resource</id>
+            <goals>
+              <goal>add-test-resource</goal>
+            </goals>
+            <configuration>
+              <resources>
+                <resource>
+                  <directory>../../main/tests/charset/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+                <resource>
+                  <directory>../../main/tests/framework/src</directory>
+                  <excludes>
+                    <exclude>**/*.java</exclude>
+                  </excludes>
+                </resource>
+              </resources>
+            </configuration>
+          </execution>
+
+
+        </executions>
+      </plugin>
+
+      <!--
+        Extract the resource files from the icu4j-datafiles dependency so that it is available during `test`.
+        Note: The extraction here will not result in these files being included in the packaged jar.
+      -->
+      <plugin>
+        <artifactId>maven-remote-resources-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>process-remote-resources</id>
+            <goals>
+              <goal>process</goal>
+            </goals>
+            <configuration>
+              <resourceBundles>
+                <resourceBundle>com.ibm.icu:icu4j-datafiles:${icu4j.version}</resourceBundle>
+                <resourceBundle>com.ibm.icu:icu4j-test-datafiles:${icu4j.version}</resourceBundle>
```

Comment:
Nitpick: can use `${project.version}` instead of `${icu4j.version}`

It is already defined, and it is Maven idiomatic
When I see it the reaction is "ah, I know what this is".
When I see `${icu4j.version}` I am wonder
"what is this? Where is it defined? Why is it different than `${project.version}`?"